### PR TITLE
refactor(animations): make `AnimationDriver.getParentElement` required

### PR DESCRIPTION
**Pull request by JoostK**
This change is a follow up to #45057 to make `AnimationDriver.getParentElement`
a required method, which allows removing the slow path for the animation
namespace insertion logic.

BREAKING CHANGE:

The `AnimationDriver.getParentElement` method has become required, so any
implementors of this interface are now required to provide an implementation
for this method. This breakage is unlikely to affect application developers,
as `AnimationDriver` is not expected to be implemented in user code.

### CODE REVIEW

**Review by jessicajaniuk — APPROVED** [review]
LGTM :cookie:

reviewed-for: public-api, fw-animations

**Review by atscott — APPROVED** [review]
reviewed-for: public-api

File: packages/animations/browser/src/render/animation_driver.ts [hunk 1]
```diff
@@ -70,7 +70,7 @@ export abstract class AnimationDriver {
    * be required to implement this method. This method is to become required in a major version of
```
Comment (AndrewKushnir) [inline packages/animations/browser/src/render/animation_driver.ts:70]: Could you please update the method description to drop last 2 sentences as they are no longer relevant after this change.

**Review by AndrewKushnir — APPROVED** [review]
LGTM, just one quick question on the getParentElement method docs.

Reviewed-for: public-api

**Comment by atscott** [comment]
Looks like presubmit found a real failure in SSR. This will require some more investigation.

**Review by dylhunn — APPROVED** [review]
reviewed-for: public-api

**Comment by atscott** [comment]
Nevermind! Looks like a flake. Passed after a rerun

**Comment by atscott** [comment]
This PR was merged into the repository by commit 59559fdbacc00bee380b1e88c79d08b6e856ebbe.

**Pull request closed without merging**